### PR TITLE
msgraph-cli: init a 1.9.0

### PR DESCRIPTION
**Pull request by nazarewk**
## Description of changes

Packages up Microsoft Graph CLI client for Nix:
- https://github.com/microsoftgraph/msgraph-cli
- https://learn.microsoft.com/en-gb/graph/cli/installation?tabs=linux

## Things done

I have never developed or packages dotnet applications before, but after failed attempt at patching release binaries I managed to get it to run smoothly using `buildDotnetModule` when needed for task at work. Can as well share it :)

- Built on platform(s)
  - [x] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- For non-Linux: Is sandboxing enabled in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
  - [ ] `sandbox = relaxed`
  - [ ] `sandbox = true`
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://github.com/NixOS/nixpkgs/blob/master/pkgs/README.md#package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [x] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [24.11 Release Notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2411.section.md) (or backporting [23.11](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2311.section.md) and [24.05](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2405.section.md) Release notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

---

Add a :+1: [reaction] to [pull requests you find important].

[reaction]: https://github.blog/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/
[pull requests you find important]: https://github.com/NixOS/nixpkgs/pulls?q=is%3Aopen+sort%3Areactions-%2B1-desc

### CODE REVIEW

**Comment by nazarewk** [comment]
Converting to draft due to encountering login errors and requiring further investigation:
- https://github.com/microsoftgraph/msgraph-cli/issues/248
- https://github.com/microsoftgraph/msgraph-cli/issues/76

**Comment by nazarewk** [comment]
I have found it to be missing `libsecret` dependency, added it as `runtimeDeps`.

For future reference:
1. I have encountered exactly this error https://github.com/microsoftgraph/msgraph-cli/issues/248
2. I have noticed I'm not using `gnome-keyring` directly, but `KeePassXC` instead so I stumbled upon https://github.com/microsoftgraph/msgraph-cli/issues/76
3. After messing with the `mgc` CLI I have stumbled upon `--debug` flag
4. `mgc login --debug` printed the whole stacktrace and debug info, which informed me it failed to find `libsecret.XXX.so`
5. I have added `libsecret` to `runtimeDeps`
6. I have logged in succesfully
7. It had persisted the data successfully in `KeePassXC` as a `MicrosoftGraph.nocae` entry

File: pkgs/by-name/ms/msgraph-cli/package.nix [hunk 1]
```diff
@@ -0,0 +1,38 @@
+{ lib
+, buildDotnetModule
```
Comment (FliegendeWurst) [inline pkgs/by-name/ms/msgraph-cli/package.nix:2]: Please format with the new formatter.

Comment (nazarewk) [reply pkgs/by-name/ms/msgraph-cli/package.nix:2]: could you elaborate? I'm a little out of loop

Comment (FliegendeWurst) [reply pkgs/by-name/ms/msgraph-cli/package.nix:2]: I think all you need to do is:
1. rebase on master
2. execute `nix-shell` in nixpkgs
3. run nixfmt <your file>

Comment (FliegendeWurst) [reply pkgs/by-name/ms/msgraph-cli/package.nix]: You can ignore the failure on deps.nix

File: pkgs/by-name/ms/msgraph-cli/package.nix [hunk 2]
```diff
@@ -0,0 +1,31 @@
+{ lib, buildDotnetModule, dotnetCorePackages, fetchFromGitHub, libsecret }:
```
Comment (FliegendeWurst) [inline pkgs/by-name/ms/msgraph-cli/package.nix:1]:
```suggestion
{
  lib,
  buildDotnetModule,
  dotnetCorePackages,
  fetchFromGitHub,
  libsecret,
}:
```

File: pkgs/by-name/ms/msgraph-cli/package.nix [hunk 3]
```diff
@@ -0,0 +1,31 @@
...
+  runtimeDeps = [ libsecret ];
+
+  passthru.updateScript = ./update.sh;
+  meta = with lib; {
+    mainProgram = "mgc";
+    description = "Microsoft Graph CLI";
+    homepage = "https://github.com/microsoftgraph/msgraph-cli";
+    license = licenses.mit;
+    maintainers = with maintainers; [ ];
+    platforms = [ "aarch64-darwin" "x86_64-darwin" "x86_64-linux" ];
```
Comment (FliegendeWurst) [inline pkgs/by-name/ms/msgraph-cli/package.nix:29]:
```suggestion
    platforms = [
      "aarch64-darwin"
      "x86_64-darwin"
      "x86_64-linux"
    ];
```

File: pkgs/by-name/ms/msgraph-cli/package.nix [hunk 4]
```diff
@@ -0,0 +1,31 @@
...
+
+  runtimeDeps = [ libsecret ];
+
+  passthru.updateScript = ./update.sh;
+  meta = with lib; {
+    mainProgram = "mgc";
+    description = "Microsoft Graph CLI";
+    homepage = "https://github.com/microsoftgraph/msgraph-cli";
+    license = licenses.mit;
+    maintainers = with maintainers; [ ];
```
Comment (FliegendeWurst) [inline pkgs/by-name/ms/msgraph-cli/package.nix:28]:
```suggestion
    maintainers = with maintainers; [ nazarewk ];
```
New packages must have a maintainer.

**Review by FliegendeWurst — CHANGES_REQUESTED** [review]

**Review by FliegendeWurst — APPROVED** [review]

**Pull request merged**